### DISCUSSION TITLE
how to check when execution has been completed

### DISCUSSION THREAD

**Discussion by deflexable** [Microsoft/node-pty#742] [Q&A] [answered]
## Environment details

- OS:
- OS version:
- node-pty version:

## Issue description

is there a method to check when an execution completes after a command was written, for example:
 ```js 
ptyInstance.write('echo test', () => {
      // execution has been completed
  })
```

**Comment by Tyriar** [comment, marked as answer]
You cannot without something like a [shell integration script](https://github.com/microsoft/vscode/blob/330ab6c2928963dd83a7d90a271d9f2eb8db90d2/src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-bash.sh)
